File: fix_question_marks.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import os
import re

# Files to fix
files_to_fix = [
    ('frontend/src/pages/ProductDetail.css', [
        (r'content: "\\?\\?";', 'content: "👤";'),
    ]),
    ('frontend/src/index.css', [
        (r'content: "\\?\\?";', 'content: "👤";'),
        (r'content: "\\?\\?\\?";', 'content: "🛍️";'),
    ]),
    ('frontend/src/pages/Explore.module.css', [
        (r"content: '\\?\\?';", "content: '🔍';"),
    ]),
    ('frontend/src/components/AffiliateTab.css', [
        (r"content: '\\?\\?';", "content: '💰';"),
    ]),
]

for filepath, replacements in files_to_fix:
    full_path = filepath
    if not os.path.exists(full_path):
        print(f"❌ File not found: {full_path}")
        continue
    
    try:
        with open(full_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        for pattern, replacement in replacements:
            content = re.sub(pattern, replacement, content)
        
        if content != original_content:
            with open(full_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"✅ Fixed: {filepath}")
        else:
            print(f"⚠️  No changes needed: {filepath}")
    except Exception as e:
        print(f"❌ Error processing {filepath}: {e}")

print("\n✨ Done!")

File: fix_question_marks.js
const fs = require('fs');
const path = require('path');

const filesToFix = [
  {
    path: 'frontend/src/pages/ProductDetail.css',
    replacements: [
      { from: /content: "\\?\\?";/g, to: 'content: "👤";' }
    ]
  },
  {
    path: 'frontend/src/index.css',
    replacements: [
      { from: /content: "\\?\\?";/g, to: 'content: "👤";' },
      { from: /content: "\\?\\?\\?";/g, to: 'content: "🛍️";' }
    ]
  },
  {
    path: 'frontend/src/pages/Explore.module.css',
    replacements: [
      { from: /content: '\\?\\?';/g, to: "content: '🔍';" }
    ]
  },
  {
    path: 'frontend/src/components/AffiliateTab.css',
    replacements: [
      { from: /content: '\\?\\?';/g, to: "content: '💰';" }
    ]
  }
];

filesToFix.forEach(({ path: filePath, replacements }) => {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    replacements.forEach(({ from, to }) => {
      content = content.replace(from, to);
    });
    
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`✅ Fixed: ${filePath}`);
    } else {
      console.log(`⚠️  No changes needed: ${filePath}`);
    }
  } catch (error) {
    console.error(`❌ Error processing ${filePath}:`, error.message);
  }
});

console.log('\n✨ Done!');
